### PR TITLE
fixes for reverse proxies

### PR DESCRIPTION
**Pull request by tardyp**
When buildbot sits behind reverse proxies:

- fonts where fetched using /fonts instead of /proxypath/fonts
- selforigin was miscaculated. /proxypath/ should be dropped from buildbotURL=http://server/proxypath -> origin is http://server

### CODE REVIEW

File: master/buildbot/test/unit/test_www_rest.py [hunk 1]
```diff
@@ -80,6 +80,23 @@ def test_invalid_http_method(self):
...
+        self.rsrc.reconfigResource(self.master.config)
+        self.assertEqual([r.pattern for r in self.rsrc.origins], ['http\\:\\/\\/server\\Z(?ms)'])
+
+        self.master.config.buildbotURL = 'http://server:8080/'
+        self.rsrc.reconfigResource(self.master.config)
+        self.assertEqual([r.pattern for r in self.rsrc.origins], ['http\\:\\/\\/server\\:8080\\Z(?ms)'])
+
+        self.master.config.buildbotURL = 'https://server:8080/'
+        self.rsrc.reconfigResource(self.master.config)
+        self.assertEqual([r.pattern for r in self.rsrc.origins], ['https\\:\\/\\/server\\:8080\\Z(?ms)'])
```
Comment (rutsky) [inline master/buildbot/test/unit/test_www_rest.py]: I'm completely out of context, but why not to use Python's raw strings?

E.g.:
```python
r'https\:\/\/server\:8080\Z(?ms)'
```

Comment (tardyp) [inline master/buildbot/test/unit/test_www_rest.py]: I usually just copy paste what assertEqual complains the value should be (after visually checking them)

**Comment by tardyp** [comment]
addressed @rutsky comments

**Comment by djmitche** [comment]
Thanks for the review @rutsky!

**Pull request merged**